pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -283,40 +271,36 @@ int sock_dtls_session_create(sock_dtls_t *sock, const sock_udp_ep_t *ep,
     memcpy(&remote->dtls_session.addr, &ep->addr.ipv6, sizeof(ipv6_addr_t));
     _ep_to_session(ep, &remote->dtls_session);
 
-    /* start a handshake */
-    DEBUG("sock_dtls: starting handshake\n");
-    res = dtls_connect(sock->dtls_ctx, &remote->dtls_session);
+    sock_udp_ep_t local;
+    if (sock_udp_get_local(sock->udp_sock, &local) < 0) {
+        return -EADDRNOTAVAIL;
+    }
+    if (ep->port == 0) {
+        return -EINVAL;
+    }
+    switch (ep->family) {
+        case AF_INET:
+ #if IS_ACTIVE(SOCK_HAS_IPV6)
+        case AF_INET6:
+ #endif
+            break;
+        default:
+            return -EINVAL;
+    }
```

Comment:
Shouldn't these checks come *before* changing `remote` in the lines before?

Reply:
yup you're right

Reply:
Still needs to be addressed as well btw ;-)

Reply:
Fixed.